scheduler/pkg/store/pipeline/utils.go

```diff
@@ -200,6 +252,29 @@ func updateInputSteps(pipelineName string, inputs []string) []string {
 	return updatedInputs
 }
 
+func updateExternalInputSteps(inputs []string) []string {
+	if len(inputs) == 0 {
+		return inputs
+	}
+	var updatedInputs []string
+	for _, inp := range inputs {
+		parts := strings.Split(inp, StepNameSeperator)
+		switch len(parts) {
+		case 1: // Add outputs if just pipeline specified
+			updatedInputs = append(updatedInputs, fmt.Sprintf("%s.%s", inp, StepOutputSpecifier))
+		case 3: // Add outputs if step name only specified
```

Comment:
as related to previous comments, I am not sure for the reader why it is 1 or 3 parts? perhaps add more description in code.